tests/lib/rules/no-unused-prop-types.js

```diff
@@ -2012,6 +2012,79 @@ ruleTester.run('no-unused-prop-types', rule, {
         '}'
       ].join('\n'),
       parser: 'babel-eslint'
+    }, {
+      // issue #1002
+      code: [
+        '  type User = {',
+        '    country: string;',
+        '  };',
+
+        '  export default ({ user }: { user: User }) => (',
+        '    <div>',
+        '      { user.country }',
+        '    </div>',
+        '  );'
+      ].join('\n'),
+      parser: 'babel-eslint'
+    }, {
+      // issue #1008
+      code: [
+        'type ObjectType = {',
+        '  usedDirectly: string;',
+        '  usedFromArray: string;',
+        '};',
+
+        'export class UseDirectly extends React.Component {',
+        '  props: {',
+        '    object: ObjectType;',
+        '  }',
+
+        '  render() {',
+        '    console.log(this.props.object.usedDirectly);',
+        '    return null;',
+        '  }',
+        '}'
+      ].join('\n'),
+      parser: 'babel-eslint'
+    }, {
+      // issue #1323
+      code: [
+        'type AProps = {',
+        '  name: string;',
+        '  age: number;',
+        '};',
+
+        'type BProps = {',
+        '  greeting: string;',
+        '};',
+
+        'type Props = AProps & BProps;',
```

Comment:
ok so this should either mean "name and age and greeting", or "name and age, or greeting" - in the former case, "age" is missing, in the latter case, "greeting" and "name" are mutually exclusive. Shouldn't this be an error?